OtpCore/OtpAuthUri.cs

```diff
@@ -76,9 +80,10 @@ public OtpAuthUri(Uri uri)
             Label = HttpUtility.UrlDecode(Uri.Segments[1]);
```

Comment:
For consistency, you'll probably want to use `Uri.UnescapeDataString()` here, so that it matches what you use in your `.ToString()` method. As you noted in your `.ToString()` method about the `HttpUtility.UrlEncode()` method, one of the biggest things it does differently is treat the plus character, `+`, as a space. This goes way back to Microsoft's Classic ASP in the late 1990's. The specifications for URIs, URLs, and `application/x-www-form-urlencoded` MIME type encoding don't all agree, hence all of the confusion and countless bugs for the last 20 years.

I'm sure anyone can argue it either way and there is no right answer that everyone can agree upon, but in your case I would just try to go for consistency within your code. Namely, if someone provides a complete URI string as the input, then your output should match that. But currently it would not if someone does the following:
```
Console.WriteLine("With a plus character.");

// A plus character is a valid character in a URI. So I expect to get this value back,
// whether it is encoded or not.
var label = "Example:my+account";

var input = $"otpauth://totp/{label}?secret=JBSWY3DPEHPK3PXP&issuer=Example";

var otp = new Petrsnd.OtpCore.OtpAuthUri(input);

Console.WriteLine($"Input:  {input}");
Console.WriteLine($"Output: {otp}");

var outUri = new Uri(otp.ToString());

var result = Uri.UnescapeDataString(outUri.Segments[1]);

Console.WriteLine();
Console.WriteLine($"Input:  {label}");
Console.WriteLine($"Output: {result}");
```
![image](https://user-images.githubusercontent.com/22459191/211917891-324f626b-ac34-4a23-9230-8fa054ee607b.png)

Reply:
Yeah. I almost went back and switched when I made the fix for the pluses that did show up, but it only happened going the one direction, not the other. I agree though. I should use the opposite function from the same class.